interpreter/terminal_interface/magic_commands.py

```diff
@@ -23,38 +32,45 @@ def handle_undo(self, arguments):
 
-    print("") # Aesthetics.
+    print("")  # Aesthetics.
 
     # Print out a preview of what messages were removed.
     for message in removed_messages:
-      if 'content' in message and message['content'] != None:
-        display_markdown_message(f"**Removed message:** `\"{message['content'][:30]}...\"`")
-      elif 'function_call' in message:
-        display_markdown_message(f"**Removed codeblock**") # TODO: Could add preview of code removed here.
-    
-    print("") # Aesthetics.
+        if 'content' in message and message['content'] != None:
+            display_markdown_message(
+                f"**Removed message:** `\"{message['content'][:30]}...\"`")
+        elif 'function_call' in message:
+            # TODO: Could add preview of code removed here.
+            display_markdown_message(f"**Removed codeblock**")
+
+    print("")  # Aesthetics.
+
 
 def handle_help(self, arguments):
     commands_description = {
-      "%debug [true/false]": "Toggle debug mode. Without arguments or with 'true', it enters debug mode. With 'false', it exits debug mode.",
-      "%reset": "Resets the current session.",
-      "%undo": "Remove previous messages and its response from the message history.",
-      "%save_message [path]": "Saves messages to a specified JSON path. If no path is provided, it defaults to 'messages.json'.",
-      "%load_message [path]": "Loads messages from a specified JSON path. If no path is provided, it defaults to 'messages.json'.",
-      "%tokens [prompt]": "Calculate the tokens used by the current conversation's messages and estimate their cost and optionally calculate the tokens and estimated cost of a `prompt` if one is provided.",
-      "%help": "Show this help message.",
+        "%debug [true/false]": "Toggle debug mode. Without arguments or with 'true', it enters debug mode. With 'false', it exits debug mode.",
+        "%reset": "Resets the current session.",
+        "%undo": "Remove previous messages and its response from the message history.",
+        "%save_message [path]": "Saves messages to a specified JSON path. If no path is provided, it defaults to 'messages.json'.",
+        "%load_message [path]": "Loads messages from a specified JSON path. If no path is provided, it defaults to 'messages.json'.",
+        "%tokens [prompt]": "Calculate the tokens used by the current conversation's messages and estimate their cost and optionally calculate the tokens and estimated cost of a `prompt` if one is provided.",
+        "%help": "Show this help message.",
+        "%upload": "open a File Dialog, and select a file to upload to the container. only used when using containerized code execution",
+        "%upload folder": "same as upload command, except you can upload a folder instead of just a file.",
+        "%upload file": "same as upload command, except you can upload a file.",
+        "%download" : "Download a file or directory given the file or folder name in the container."
```

Comment:
`%download` without any path has some unexpected behavior:

![Screen Shot 2023-10-23 at 5 26 07 PM](https://github.com/KillianLucas/open-interpreter/assets/1667415/a50ac8e6-8e67-4be8-9d85-db360b7b6e9c)

Reply:
oh wow. i guess it tried to download the entire /mnt/data dir. i would say thats wierd, but its really just me being blind. simple fix.